src/vm/node_modules/VM.js

```diff
@@ -11178,8 +11201,16 @@ exports.reprovision = function (uuid, payload, options, callback)
                     zfs(['mount', vmobj.zfs_filesystem], log,
                         function (err, fds) {
                             log.warn({err: err, fds: fds},
-                                '(retry) mount results');
-                            cb(err);
+                            '(retry + ' + (retry_count + 1) + ') mount '
+                                + 'results');
+
+                            retry_count++;
+
+                            if (err && retry_count < retry_max) {
+                                _retryBusy();
+                            } else {
+                                cb(err);
+                            }
```

Comment:
Given that the symptom is "dataset busy", and the max retries is 10 (i.e., 10 seconds), maybe the best way to handle this is to simply retry 10s later instead of 1s rather than having to deal with retry counts.

Just changing `retry_delay` is a lot less invasive a change that I would feel more comfortable with.